tests/behat/app_behat_runtime.js

```
@@ -98,15 +98,20 @@
         var index = requestIndex++;
         var key = 'httprequest-' + index;
 
-        // Add to the list of pending requests.
-        addPending(key);
+        try {
+            // Add to the list of pending requests.
+            addPending(key);
 
-        // Detect when it finishes and remove it from the list.
-        this.addEventListener('loadend', function() {
-            removePending(key);
-        });
+            // Detect when it finishes and remove it from the list.
+            this.addEventListener('loadend', function() {
+                removePending(key);
+            });
 
-        return realOpen.apply(this, arguments);
+            return realOpen.apply(this, arguments);
+        } catch (e) {
+            removePending(key);
+            return false;
```

In order to keep the default behavior in case of an error, could we throw the error again after removing the pending request instead of returning false? I think it should also solve the issue you mentioned.